### PR TITLE
WIP: Elekta xvi

### PR DESCRIPTION
**Pull request by adler-j**
Need to add doc and some distinction on offset detector etc.

### CODE REVIEW

**Comment by adler-j** [comment]
Now ready for a review, offset detector will have to come later.

**Comment by adler-j** [comment]
Could use a review so I can get this in

File: odl/contrib/tomo/elekta.py [hunk 1]
```diff
@@ -190,6 +193,158 @@ def elekta_icon_fbp(ray_transform,
...
+def elekta_xvi_space(shape=(512, 512, 512), **kwargs):
+    """Default reconstruction space for the Elekta XVI CBCT.
+
+    Parameters
+    ----------
+    shape : sequence of int, optional
+        Shape of the space, in voxels.
+    kwargs :
+        Keyword arguments to pass to `uniform_discr` to modify the space, e.g.
+        use another backend.
```
Comment (kohr-h) [inline odl/contrib/tomo/elekta.py]: Perhaps mention primarily `dtype`, and mention that the default in float32.

File: odl/contrib/tomo/elekta.py [hunk 2]
```diff
@@ -190,6 +193,158 @@ def elekta_icon_fbp(ray_transform,
     return fbp_op
 
 
+def elekta_xvi_geometry(sad=1000.0, sdd=1500.0,
+                        piercing_point=(512.0, 512.0),
+                        angles=np.linspace(0, 2 * np.pi, 650),
+                        detector_shape=(1024, 1024)):
+    """Tomographic geometry of the Elekta XVI system.
+
+    All measurments are given in millimeters unless otherwise stated.
```
Comment (kohr-h) [inline odl/contrib/tomo/elekta.py]: Perhaps there's a reference (like a link) to the system description that you can provide?

File: odl/contrib/tomo/elekta.py [hunk 3]
```diff
@@ -190,6 +193,158 @@ def elekta_icon_fbp(ray_transform,
...
+    ----------
+    ray_transform : `RayTransform`
+        The ray transform to be used, should have an Elekta XVI geometry.
+    padding : bool, optional
+        If the fbp filter should use padding, increases memory use
+        significantly.
+    filter_type : str, optional
+        Type of filter to apply in the FBP filter.
+    frequency_scaling : float, optional
+        Frequency scaling for FBP filter
```
Comment (kohr-h) [inline odl/contrib/tomo/elekta.py]: .

File: odl/contrib/tomo/elekta.py [hunk 4]
```diff
@@ -190,6 +193,158 @@ def elekta_icon_fbp(ray_transform,
...
+def elekta_xvi_fbp(ray_transform,
+                   padding=False, filter_type='Hann', frequency_scaling=0.6):
+    """Approximation of the FDK reconstruction used in the Elekta XVI.
+
+    Parameters
+    ----------
+    ray_transform : `RayTransform`
+        The ray transform to be used, should have an Elekta XVI geometry.
+    padding : bool, optional
+        If the fbp filter should use padding, increases memory use
```
Comment (kohr-h) [inline odl/contrib/tomo/elekta.py]: Whether
FBP

File: odl/contrib/tomo/examples/elekta_icon_algebraic_reconstruction.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+"""Example of using the Elekta Icon geometry.
+
+In this example we create an Elekta Icon geometry and use it to create some
+artificial data and reconstruct it using ART, specifically kaczmarz method.
```
Comment (kohr-h) [inline odl/contrib/tomo/examples/elekta_icon_algebraic_reconstruction.py]: Kaczmarz' method

File: odl/contrib/tomo/examples/elekta_icon_algebraic_reconstruction.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+"""
+
+import numpy as np
+import odl
+from odl.contrib import tomo
+
+subsets = 20
```
Comment (kohr-h) [inline odl/contrib/tomo/examples/elekta_icon_algebraic_reconstruction.py]: A word on this?

File: odl/contrib/tomo/examples/elekta_xvi_fbp.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+# Display the results
+phantom.show('phantom xz', coords=[None, 0, None])
+reconstruction.show('reconstruction yz', coords=[None, 0, None])
+
+phantom.show('phantom xy', coords=[None, None, 0])
+reconstruction.show('reconstruction yz', coords=[None, None, 0])
+
+phantom.show('phantom yz', coords=[0, None, None])
+reconstruction.show('reconstruction yz', coords=[0, None, None])
```
Comment (kohr-h) [inline odl/contrib/tomo/examples/elekta_xvi_fbp.py]: The titles of the reco images are mixed up (also in the other FBP example)

File: odl/contrib/tomo/examples/elekta_xvi_fbp.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+# Get default geometry and space
+geometry = tomo.elekta_xvi_geometry()
+space = tomo.elekta_xvi_space(shape=(112, 112, 112))
+
+# Create ray transform
+ray_transform = odl.tomo.RayTransform(space, geometry,
+                                      use_cache=False)
+
+# Get default FDK reconstruction
```
Comment (kohr-h) [inline odl/contrib/tomo/examples/elekta_xvi_fbp.py]: operator

File: odl/contrib/tomo/examples/elekta_icon_algebraic_reconstruction.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+# Create sub-geometries using geometry indexing
+step = int(np.ceil(geometry.angles.size / subsets))
+geometries = [geometry[i * step:(i+1) * step] for i in range(subsets)]
+
+# Create ray transform
+ray_transforms = [odl.tomo.RayTransform(space, geom, use_cache=False)
+                  for geom in geometries]
+
+# Create simplified phantom
```
Comment (kohr-h) [inline odl/contrib/tomo/examples/elekta_icon_algebraic_reconstruction.py]: simplified?

File: odl/contrib/tomo/examples/elekta_xvi_algebraic_reconstruction.py [hunk 10]
```diff
@@ -0,0 +1,36 @@
+"""Example of using the Elekta XVI geometry.
+
+In this example we create an Elekta XVI geometry and use it to create some
+artificial data and reconstruct it using ART, specifically kaczmarz method.
+"""
+
```
Comment (kohr-h) [inline odl/contrib/tomo/examples/elekta_xvi_algebraic_reconstruction.py]: Same comments as in other ART example

File: odl/contrib/tomo/elekta.py (hunk 2 above)
Comment (adler-j) [reply odl/contrib/tomo/elekta.py]: It seems not, I'm trying to find one. This is only mentioned in derived articles (e.g. some medical physisist has measured it)

File: odl/contrib/tomo/elekta.py [hunk 11]
```diff
@@ -190,6 +193,158 @@ def elekta_icon_fbp(ray_transform,
     return fbp_op
 
 
+def elekta_xvi_geometry(sad=1000.0, sdd=1500.0,
+                        piercing_point=(512.0, 512.0),
+                        angles=np.linspace(0, 2 * np.pi, 650),
```
[the remark this answers is not in the document]
Comment (adler-j) [reply odl/contrib/tomo/elekta.py]: I'm actually happy with changing these to `None` by default, will do.

File: odl/contrib/tomo/elekta.py (hunk 2 above)
Comment (kohr-h) [reply odl/contrib/tomo/elekta.py]: Maybe this:
https://www.elekta.com/radiotherapy/treatment-solutions/imaging/xvi.html
??

Comment (adler-j) [reply odl/contrib/tomo/elekta.py]: Says nothing about the geometry :/

Comment (kohr-h) [reply odl/contrib/tomo/elekta.py]: That's harder of course. Still nice to have something to click on, better than nothing.

**Comment by adler-j** [comment]
Merge after CI

**Pull request merged**